lib/utils/getlogger.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * @returns {Logger}
+ */
+module.exports = function getLogger() {
+	const logsCollection = new Map( [
+		[ 'info', [] ],
+		[ 'error', [] ]
+	] );
+
+	let lastInserted;
+
+	const logger = {
+		/**
+		 * @param {String} msg
+		 */
+		info( msg ) {
+			return logger._log( 'info', msg );
+		},
+
+		/**
+		 * @param {String|Error} msg
+		 */
+		error( msg ) {
+			if ( msg instanceof Error ) {
+				msg = msg.stack;
+			}
+
+			return logger._log( 'error', msg );
+		},
+
+		/**
+		 * @param {Logs} responseLogs
+		 */
+		concat( responseLogs ) {
+			responseLogs.info.forEach( ( msg ) => logger.info( msg ) );
+
+			responseLogs.error.forEach( ( err ) => logger.error( err ) );
+		},
+
+		/**
+		 * @returns {Logs}
+		 */
+		all() {
+			return {
+				error: logsCollection.get( 'error' ),
+				info: logsCollection.get( 'info' ),
+			};
+		},
+
+		/**
+		 * @returns {String|undefined}
+		 */
+		getLastInserted() {
```

Comment:
You do not insert docs. It's an implementation detail what this util does with the logs.